assets/js/googlesitekit/widgets/components/WidgetAreaHeader.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WidgetAreaHeader component.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useWindowWidth } from '@react-hook/window-size/throttled';
+
+/**
+ * Internal dependencies
+ */
+import { Cell, Row } from '../../../material-components';
+import { useSelect } from 'googlesitekit-data';
+import { CORE_WIDGETS } from '../datastore/constants';
+import WidgetNewBadge from './WidgetNewBadge';
+
+export default function WidgetAreaHeader( { slug } ) {
+	const widgetArea = useSelect( ( select ) =>
+		select( CORE_WIDGETS ).getWidgetArea( slug )
+	);
+
+	const { Icon, title, subtitle, CTA } = widgetArea;
```

Comment:
See the comments bellow, if we pass these as props directly, we do not need to re-import them from widget area selector, which is already handled in parent component

Reply:
@zutigrm addressed in the latest commit.